### PR TITLE
Privacy Policy / GDPR / Code of Conduct Footer / Minor Changes

### PR DESCRIPTION
**Pull request by snaeker58**
TODO: Actual code of conduct, making the <br>/<strong> element thicc.</strong>

The Code of Conduct is based off the Contributor Covenant, which we were using previously.

### CODE REVIEW

File: src/lib/Page.svelte [hunk 1]
```diff
@@ -11,27 +11,27 @@
 </div>
 
 <style>
-  div {
-    width: 1024px;
-    margin: auto;
-    padding: 8em 2em;
-    max-width: 100%;
-  }
+    div {
```
Comment (iamawatermelo) [inline src/lib/Page.svelte]: Please reconfigure your editor settings to use two space indents

File: src/routes/+layout.svelte [hunk 2]
```diff
@@ -59,7 +59,10 @@
       <div class="links">
         <p class="title">Legal</p>
         <ul>
+          <!-- TODO: Add link to Code of Conduct (which does not exist yet) -->
+          <li><a href="/404">Code of Conduct</a></li>
           <li><a href="/privacy">Privacy policy</a></li>
+          <li><a href="/privacy/gdpr">GDPR Compliance</a></li>
```
Comment (iamawatermelo) [inline src/routes/+layout.svelte]: Please use sentence case

File: src/routes/privacy/+page.svelte [hunk 3]
```diff
@@ -3,88 +3,115 @@
...
 
 <Page title="Privacy">
-  <h1>Privacy</h1>
+  <h1>Privacy Policy</h1>
 
-  <p>This privacy notice tells you what to expect us to do with your personal information.</p>
+  <p>This privacy policy explains how the Zenith Hacks Team (referred to as "we" or "us") handles your personal
+    information. For residents of the European Union (EU) and California, please also refer to our <a
+      href="/privacy/gdpr">GDPR</a> and <a
+      href="/privacy/ccpa">CCPA</a> Compliance
```
Comment (iamawatermelo) [inline src/routes/privacy/+page.svelte]: This page  does not exist?

File: src/routes/privacy/+page.svelte [hunk 4]
```diff
@@ -3,88 +3,115 @@
...
-    We also store the following information:
+    We collect this information to:
   </p>
   <ul>
-    <li>your IP address</li>
-    <li>your approximate location (the city and country we think you live in)</li>
-    <li>the time of day you sent the request</li>
+    <li>Notify you about the opening of sign-ups.</li>
+    <li>Ensure the security of our systems and prevent abuse or spam.</li>
+    <li>Improve the quality of our services based on user patterns and preferences.</li>
```
Comment (iamawatermelo) [inline src/routes/privacy/+page.svelte:66]: We do not collect data for analytics purposes

File: src/routes/privacy/+page.svelte [hunk 5]
```diff
@@ -3,88 +3,115 @@
...
   </p>
 
-  <h2 id="retention">How long do we keep your data?</h2>
+  <h2 id="retention">Data Retention Period</h2>
   <p>
-    When you ask to be notified when sign-ups open using the form on our website, we keep the information that you give
-    us until we send you that email and for no longer than 12 months.<br />
-    After then, we delete the information.
+    We retain your personal information in Google Cloud servers for up to 12 months, as outlined in the <a
+    href="#information-collected">Information
```
Comment (iamawatermelo) [inline src/routes/privacy/+page.svelte]: Maintain case for anchor links please, do not capitalise

File: src/routes/privacy/+page.svelte [hunk 6]
```diff
@@ -3,88 +3,115 @@
...
+    We may share your information with the following entities. Note that these external organizations have their own
+    privacy policies, and we are not responsible for their actions:
+  </p>
   <ul>
-    <li>
-      Cloudflare, which hosts our website and processes your information. Their DPA is <a
-        href="https://www.cloudflare.com/en-gb/cloudflare-customer-dpa/">here</a
-      >, and by using our services you agree to it.
+    <li><b>Google:</b> We store your data on Google Cloud servers. By accepting our privacy policy you also consent to
+      <a href="https://cloud.google.com/terms/cloud-privacy-notice">Google Clouds privacy policy</a>.
```
Comment (iamawatermelo) [inline src/routes/privacy/+page.svelte]: Google Cloud's privacy policy is for GCP customers. The previous DPA was the correct link.

File: src/routes/privacy/gdpr/+page.svelte [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+    We take data security seriously and employ industry-standard measures to protect your personal information from
+    unauthorized access, misuse, or disclosure. Your data is stored on secure Google servers, <a
+    href="https://cloud.google.com/privacy/gdpr">which comply with GDPR
+    regulations </a>. We do not share your personal data with third parties unless required by law or with your explicit
+    consent.
+  </p>
+
+  <h2>Your Rights Under GDPR</h2>
+  <p>
+    As a data subject under the GDPR, you have the following rights regarding your personal data:
```
Comment (iamawatermelo) [inline src/routes/privacy/gdpr/+page.svelte]: Not everyone is in the EU: "If you live in a country where the GDPR applies,"

**Review by iamawatermelo — CHANGES_REQUESTED** [review]
The tone is completely wrong. I understand it is a legal document, but we should aim to make it as understandable as possible because we deal with minors. ICO guidance is to make all legal text as easy to read as possible if your service is provided to minors.

Please rewrite in plain English.

See guidance on [the right to be informed](https://ico.org.uk/for-organisations/uk-gdpr-guidance-and-resources/childrens-information/children-and-the-uk-gdpr/how-does-the-right-to-be-informed-apply-to-children/#:~:text=to%20be%20informed-,How%20should%20we%20provide%20privacy%20information%3F,use%20of%20dashboards%2C%20layers%2C%20just%2Din%2Dtime%20notices%2C%20icons%20and%20symbols.,-What%20if%20we).

File: src/routes/privacy/gdpr/+page.svelte (hunk 7 above)
Comment (snaeker58) [reply src/routes/privacy/gdpr/+page.svelte]: That is stated in the Privacy Policy!

File: src/routes/privacy/+page.svelte (hunk 6 above)
Comment (snaeker58) [reply src/routes/privacy/+page.svelte]: My mistake!

File: src/routes/privacy/+page.svelte (hunk 4 above)
Comment (snaeker58) [reply src/routes/privacy/+page.svelte:66]: We do not collect analytics data, we do might indirectly use / extrapolate from the few bits of data we do store from the email sign-up to change something on the site which falls under that 3rd category.

File: src/routes/privacy/+page.svelte (hunk 5 above)
Comment (snaeker58) [reply src/routes/privacy/+page.svelte]: Case is maintained as titles get capitalised no matter if they link or not.

File: src/lib/Page.svelte (hunk 1 above)
Comment (snaeker58) [reply src/lib/Page.svelte]: Will be fixed!

File: src/routes/+layout.svelte (hunk 2 above)
Comment (snaeker58) [reply src/routes/+layout.svelte]: Will be fixed!

File: src/routes/privacy/+page.svelte (hunk 3 above)
Comment (snaeker58) [reply src/routes/privacy/+page.svelte]: Will be fixed!

File: src/routes/privacy/+page.svelte (hunk 6 above)
Comment (snaeker58) [reply src/routes/privacy/+page.svelte]: Will be fixed!

Comment (snaeker58) [reply src/routes/privacy/+page.svelte]: I would use: https://cloud.google.com/terms/data-processing-addendum?hl=en

**Comment by snaeker58** [comment]
> The tone is completely wrong. I understand it is a legal document, but we should aim to make it as understandable as possible because we deal with minors. ICO guidance is to make all legal text as easy to read as possible if your service is provided to minors.
> 
> Please rewrite in plain English.
> 
> See guidance on [the right to be informed](https://ico.org.uk/for-organisations/uk-gdpr-guidance-and-resources/childrens-information/children-and-the-uk-gdpr/how-does-the-right-to-be-informed-apply-to-children/#:~:text=to%20be%20informed-,How%20should%20we%20provide%20privacy%20information%3F,use%20of%20dashboards%2C%20layers%2C%20just%2Din%2Dtime%20notices%2C%20icons%20and%20symbols.,-What%20if%20we).

I disagree, please provide me with examples. You can easily determine what we collect, for what we use what we collect and how to delete it. You can also easily find out who we share the data with.

Most people that look at the privacy policy want to find a contact email for deletion. I hate when this is hard to find, but navigation to that section is easy and it is highlighted by the fact that it is separate.

**Comment by snaeker58** [comment]
I am ready for another judging Sarah!!

**Comment by snaeker58** [comment]
Please get rid of we'll, it does not make it easier to read but sound more informal.

File: src/routes/privacy/+page.svelte [hunk 8]
```diff
@@ -5,90 +5,91 @@
...
   </ul>
+  <p>
+    We'll delete all of your data securely shortly after we have sent the email you signed up for, and no later than 12 months since you gave us the data, as outlined in the <a href="#retention">how long we keep your data for</a> section.
+  </p>
 
-  <h2 id="why-collect">Why We Collect Your Information</h2>
+  <h2 id="why-collect">Why we collect your information</h2>
   <p>
-    We collect this information to:
+    We collect the information you give us to:
```
Comment (snaeker58) [inline src/routes/privacy/+page.svelte]: Literally just longer phrasing

File: src/routes/privacy/+page.svelte [hunk 9]
```diff
@@ -5,90 +5,91 @@
...
-    When you opt in to receive notifications via email using the form on our website, we collect, store, and process the
+    When you opt in to receive a notification via email using the form on our website, we collect, store, and process the
     following information:
   </p>
   <ul>
     <li><b>Email address:</b> Used to notify you when sign-ups open.</li>
-    <li><b>IP address:</b> For security purposes and spam prevention.</li>
-    <li><b>Approximate location:</b> The city and country inferred from your IP address.</li>
-    <li><b>Time of request:</b> To track and manage your request.</li>
+    <li><b>IP address, approximate location and time of request:</b> For security purposes and spam prevention.</li>
```
Comment (snaeker58) [inline src/routes/privacy/+page.svelte]: I honestly think this makes it harder to read instead of having each thing we collect listed on its own line. Also when flying over the PP it is more easily overlooked.

**Comment by iamawatermelo** [comment]
I already said this to you but putting it here so it is public:

> We are not the government nor some Super Serious Organisation and I think our documents should reflect that. We're literally just a hackathon organised by a bunch of teens

> I'm probably going to put that in the brand guidelines

File: src/routes/privacy/+page.svelte [hunk 10]
```diff
@@ -5,90 +5,91 @@
...
-      authorities when required by law or for safeguarding reasons. This does not include private entities.
+      authorities when required by law or for safeguarding reasons.
     </li>
     <li><b>With Your Consent:</b> Your data may be shared publicly on our website, social media, or other
-      marketing channels if you have explicitly given us consent.
+      marketing channels if you have told us you consent to this.
     </li>
   </ul>
+  <p>
+    We may add or remove subprocessors at any time. They are under the same obligation as us to keep your data secure and to delete it when you or we request them to.
```
Comment (snaeker58) [inline src/routes/privacy/+page.svelte]: Please add that we are not responsible for subprocessors actions

File: src/routes/privacy/+page.svelte [hunk 11]
```diff
@@ -5,90 +5,91 @@
...
+  </p>
+  
+  <h2 id="retention">How long we keep your data for</h2>
+  <p>
+    We keep your personal information with our subprocesses for up to 12 months, as outlined in the <a
+    href="#information-collected">information we collect</a> section. However, we may delete your data sooner than this. After this period,
```
Comment (snaeker58) [inline src/routes/privacy/+page.svelte]: At least capitalize the I ffs, we're citing a title Sarah!!!

Comment (snaeker58) [inline src/routes/privacy/+page.svelte]: Cut the we may delete sooner..., it's pointless to talk in hypotheticals in a legal document

**Review by snaeker58 — COMMENTED** [review]
Actually much better than your og PP :D

File: src/routes/privacy/+page.svelte (hunk 10 above)
Comment (iamawatermelo) [reply src/routes/privacy/+page.svelte]: But we kind of are

File: src/routes/privacy/+page.svelte (hunk 11 above)
Comment (iamawatermelo) [reply src/routes/privacy/+page.svelte]: But people read it too

Comment (snaeker58) [reply src/routes/privacy/+page.svelte]: Ok so "at the latest within" or something like that?

File: src/routes/privacy/+page.svelte (hunk 10 above)
Comment (snaeker58) [reply src/routes/privacy/+page.svelte]: Well only to a certain degree, more so under GDPR:

While we carefully select our subprocessors and take many steps to ensure that they comply with relevant data protection regulations, we do not assume responsibility for the actions, omissions, or conduct of these subprocessors, besides those placed on us by the authorities that govern the laws that apply to you, us and the affected subprocessor.

Do your we're too cool to sound like suits magic!

File: src/routes/privacy/+page.svelte (hunk 9 above)
Comment (iamawatermelo) [reply src/routes/privacy/+page.svelte]: I agree. I'll change this later

File: src/routes/privacy/+page.svelte (hunk 10 above)
Comment (iamawatermelo) [reply src/routes/privacy/+page.svelte]: Okay, thanks, will update

**Comment by snaeker58** [comment]
Added the code of conduct now! Take a look when you have time if you're happy with the layout

**Review by iamawatermelo — APPROVED** [review]

**Comment by leowilkin** [comment]
I've seen this and will review soon.

**Comment by skyfallwastaken** [comment]
@leowilkin Mind taking another look at this? LGTM but a second opinion would be great
(the commit below is me fixing the merge conflicts)

**Comment by skyfallwastaken** [comment]
that was a bit random :p
